packages/vue-router/src/router.ts

```diff
@@ -212,59 +239,83 @@ export const createIonRouter = (opts: IonicVueRouterOptions, router: Router) =>
-
-          routeInfo.lastPathname = currentRouteInfo?.pathname || routeInfo.lastPathname;
-          routeInfo.pushedByRoute = pushedByRoute;
-          routeInfo.routerDirection = currentRouteInfo?.routerDirection || routeInfo.routerDirection;
-          routeInfo.routerAnimation = currentRouteInfo?.routerAnimation || routeInfo.routerAnimation;
-          routeInfo.prevRouteLastPathname = currentRouteInfo?.lastPathname;
+          nextRouteInfo.lastPathname = leavingRouteInfo.pathname;
+          nextRouteInfo.tab = leavingRouteInfo.tab;
+          nextRouteInfo.pushedByRoute = (leavingRouteInfo.pathname !== '') ? leavingRouteInfo.pathname : '/';
+        } else {
+          nextRouteInfo.routerAnimation = leavingRouteInfo?.routerAnimation;
+          nextRouteInfo.lastPathname = leavingRouteInfo?.pathname;
+          nextRouteInfo.pushedByRoute = (leavingRouteInfo.pathname !== '') ? leavingRouteInfo.pathname : '/';
         }
+      } else if (nextRouteInfo.routerAction === 'pop') {
+        /**
+         * Pop can occur by the user going from page -> page/a -> page/b -> page/a
+         *
+         * Going back from page/b to page/a would be a `pop` action
+         */
 
+        nextRouteInfo.routerAnimation = leavingRouteInfo?.routerAnimation;
+        nextRouteInfo.lastPathname = leavingRouteInfo?.pathname;
+        //somehow the pushed by route gets lost sometimes, recover original
+        if (nextRouteInfo.pushedByRoute === undefined) {
+          const oldRouteInfo = locationHistory.current(initialHistoryPosition, currentHistoryPosition);
+          nextRouteInfo.pushedByRoute = oldRouteInfo.pushedByRoute;
+        }
+      } else if (nextRouteInfo.routerAction === 'push' && isNewTab) {
+        nextRouteInfo.pushedByRoute = (leavingRouteInfo.pathname !== '') ? leavingRouteInfo.pathname : '/';
+        nextRouteInfo.lastPathname = leavingRouteInfo.pathname;
+      } else if (nextRouteInfo.routerAction === 'replace') {
+        /**
+         * Replace occurs when replacing a route, page -> page/a -(replace)-> page/b
+         * The history stack would have page -> page/b as page/a was replaced with page/b
+         */
+        const lastRouteInfo = getLeavingRouteInfo();
+
+        /**
+         * Should never be '/'
+         */
+        nextRouteInfo.replacedRoute = (leavingRouteInfo.pathname !== '') ? leavingRouteInfo.pathname : undefined;
+
+        nextRouteInfo.lastPathname = lastRouteInfo?.lastPathname;
+        nextRouteInfo.pushedByRoute = lastRouteInfo?.pushedByRoute;
+        nextRouteInfo.routerDirection = lastRouteInfo?.routerDirection;
+        nextRouteInfo.routerAnimation = lastRouteInfo?.routerAnimation;
       }
 
-      routeInfo.position = currentHistoryPosition;
-      routeInfo.delta = delta;
+      //
+      nextRouteInfo.matchedPath = location.matched[0].path;
+      nextRouteInfo.position = currentHistoryPosition;
+      //why store the delta of this route info?
+      // nextRouteInfo.delta = delta;
```

Comment:
Delta is stored for the page transition logic (introduced in this PR: #24056). Liam can probably describe the exact implementation reasoning in more detail.

Reply:
I restructured the IonRouterOutlet to be able to handle all transitions from all outlets. Since it is impossible from one outlet to find all the entering/leaving viewItems you have to let all outlets handle transitions. So I think I may have removed the requirement for this.

The use case was if a user went from page/a  -> tab/1 -> tab/2 -> page/b. Since the tabs were on a different outlet, page/b could not find the tab/2 to navigate away from, so life cycles never fired and it was never hidden, the outlet it was in was hidden.